fmriprep/config.py

```diff
@@ -401,12 +404,14 @@ def init(cls):
         if cls._layout is None:
             import re
             from bids.layout import BIDSLayout
-            work_dir = cls.work_dir / 'bids.db'
-            work_dir.mkdir(exist_ok=True, parents=True)
+            _db_path = cls.database_path or (cls.work_dir / cls.run_uuid / 'bids.db')
```

Comment:
Is `bids.db` meant to be a folder? (because bidslayout will create a folder)

Reply:
That said, keeping an index under the `run_uuid` folder will make the caching useless in 99% of the cases, as normally a new uuid will be generated. I would rather keep the general database under work_dir.

Reply:
> Is `bids.db` meant to be a folder? (because bidslayout will create a folder)

yes, i kept the name you used in your original implementation.

> That said, keeping an index under the `run_uuid` folder will make the caching useless in 99% of the cases, as normally a new uuid will be generated. I would rather keep the general database under work_dir.

Well, there's a trade off there too. If we generate at the top-level `work_dir`, we have to handle potential race conditions of database index initializing in cases where multiple participant labels are run concurrently.

Reply:
> Well, there's a trade off there too. If we generate at the top-level `work_dir`, we have to handle potential race conditions of database index initializing in cases where multiple participant labels are run concurrently.

You mean in NFS systems? I believe directory creation is the only atomic operation in NFS systems - meaning it is either present or not present.

What about the following:

- if the database dir exists in the work_dir, it passed in to BIDSLayout as is and not forced to reindex.
- if the database dir does not exist, can we initialize it to the run_uuid folder and then move to the work_dir (except some other process did already and it exists)?

Reply:
I think asking people to pre-create a database cache if they want to use it without re-indexing (and thus from multiple concurrent runs) is fine. The alternative seems to be asking for subtle bugs.

Reply:
after talking about this with @oesteban, we agreed it would be nice to provide a CLI tool to generate a database index of a given BIDS dataset. @effigies do you think that would be something we could/should include in pybids, or rather in another spinoff package?

Reply:
That could easily go in pybids, but it would need to be flexible. An index depends on initialization parameters, so if two apps use different initialization parameters, a user would need to know enough to initialize two separate indices.